### ISSUE TITLE
persistence

### ISSUE THREAD

**Issue by thomasdelaet** [kadtools/kad-telemetry#7]
would it be possible to use the kad-fs interface for persistence on this one?

**Comment by gordonwritescode** [comment]
Yes, i think it would be much better if we could use the same pluggable storage adapters here. It should be pretty easy to adapt - the persistence module is quite thin.

Unfortunately I don't have the time right now to work on this (though I will need it in the coming weeks for [storj](https://github.com/Storj/node-storj) and [metadisk](https://github.com/Storj/metadisk-api)). In the meantime, I'd happily accept a pull request. :smiley:

**Comment by thomasdelaet** [comment]
I will but will take a few weeks ...

**Comment by thomasdelaet** [comment]
Took a bit longer than a few weeks but I created a first cut at this. Note that his:
a) does not yet adapt tests
b) untested in actual deployment
c) relies on async getNearestContacts (PR in main kad repo)

**Comment by thomasdelaet** [comment]
would be great to get feedback if this goes conceptually in the right direction
